shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/controllers/BridgeExecutorController.java

```diff
@@ -59,9 +59,9 @@ public void init(EventSourceManager eventSourceManager) {
             LOGGER.debug("Executor deployment BridgeProcessor: '{}' in namespace '{}' is NOT ready", bridgeExecutor.getMetadata().getName(),
                     bridgeExecutor.getMetadata().getNamespace());
 
-            // TODO: Check if the deployment is in an error state, update the CRD and notify the manager!
+            // TODO: notify the manager if in FailureState: .status.Type = Ready and .status.Reason = DeploymentFailed
```

Comment:
Hi @ricardozanini , do we have `Ready` when there is a failure as well as everything is running fine?

Reply:
Hi! The Conditions are already there, they change based on the current condition of the object. For example, if we have a failure, we could have the following:

| Type | Status | Reason                     | Message |
| ------ | ------ | -------------------|----------|
| Ready | False  | DeploymentFailure | Image not found (whatever error message) |
| Augmentation | False | | |

This means that the object is in failure condition, not ready, and not processing the resource adapter (in this case the Deployment).

Of course, we can potentially have much more conditions in the long run or refactor the way we do today. The important thing is the course of action and the structure of the conditions to be compatible with the rest of the objects in the cluster.